src/constructor/load.py

```diff
@@ -0,0 +1,220 @@
+            weight: Added weight.
+            depth: Added depth.
+        """
+        self.__state_dict[key] = weight
+        self.__state_dict_depth[key] = depth
+
+    def add(self, key: str, weight: torch.Tensor, depth: int):
+        """Add weight and depth to dicts by key if current depth more than previous.
+        
+        Args:
+            key: Added weight key.
+            weight: Added weight.
+            depth: Added depth.
+        """
+        if key in self.__state_dict_depth:
+            old_depth = self.__state_dict_depth[key]
+            if depth > old_depth:
+                self.__add(key, weight, depth)
+        else:
+            self.__add(key, weight, depth)
+        
+    @property
+    def state_dict(self) -> Dict[str, torch.Tensor]:
+        return self.__state_dict
+
+    @property
+    def state_dict_depth(self) -> Dict[str, int]:
+        return self.__state_dict_depth
+
+    
+def create_override_state_dict(override_name2state_dict: Dict[str, Dict[str, torch.Tensor]], 
+                               model_keys: List[str]) -> Dict[str, torch.Tensor]:
+    """Create one state dict from dictionary of override state dicts.
+    
+    If override state dicts have many weights of one straighten key then the weight with the biggest depth
+    would be written in final state dict. There is the depth is the count of dots in the dictionary key.
+
+    Example:
+        >>> backbone = {
+        >>>     'backbone.linear.1': torch.tensor(5),
+        >>>     'backbone.linear.2': torch.tensor(3)
+        >>> }
+        >>> backbone_linear_1 = {
+        >>>     'backbone.linear.1': torch.tensor(10)
+        >>> }
+        >>> override_name2state_dict = {
+        >>>     'backbone': backbone,
+        >>>     'backbone.linear.1': backbone_linear_1
+        >>> }
+        >>> create_override_state_dict(override_name2state_dict)
+        {'backbone.linear.1': tensor(10), 'backbone.linear.2': tensor(3)}
+
+    Args: 
+        override_name2state_dict: Dictionary of module name to it's state_dict.
+        model_keys: Load model keys.
+
+    Returns:
+        override_state_dict: Created state dictionary.
+
+    Raises:
+        ValueError: If override checkpoint key not in model keys.
+    """
+    state_dict_with_depth = StateDictWithDepthStructure()
+    for key, state_dict in override_name2state_dict.items():
+        straighten_keys = create_straighten_keys(key, model_keys)
+        depth = len(key.split('.'))
+        if len(straighten_keys) == 0:
+            raise ValueError(f'Error in load checkpoint function. '
+                             f'Key `{key}` in ovrride checkpoint not in model state dict')
+        for straighten_key in straighten_keys:
+            prefix_len = len(key) + 1
+            override_key = straighten_key[prefix_len: ]
+            # If override_key not in state dict i.e override checkpoint not contain full weights for module
+            if override_key not in state_dict:
```

Comment:
How this can happen? If the keys are straightened, then you shouldn't even try to find a part of the key in state_dict: `if override_key not in state_dict`